syntax/grammar

```diff
@@ -0,0 +1,52 @@
+
+use_statement: identifier ['as' identifier]
+
+function_def: 'fn' NAME '(' decl_list ')' type NEWLINE function_body
+
+return_statement: 'ret' [expression]
+
+decl_list: decl | decl "," decl_list | epsilon
+
+type: i64 | i32 | i16 | i8 | i1 | f32 | f64 | str | '[' type ']' | void | NAME
```

Comment:
Widths are any arbitrary number >= 1. `i1`, `i2`, `i32`, `i99999` are all valid. `i0` and `i-1`, for example, are not.